src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2468,71 +2603,241 @@ private void CutOrDelete(IEnumerable<SnapshotSpan> projectionSpans, bool isCut)
+                    {
+                        // don not delete and paste anything if any part of selection is not in current submission
+                        if (!IsSelectionInsideCurrentSubmission())
+                        {
+                            return false;
+                        }
+
+                        DeleteSelection();
+
+                        if (selection.Mode == TextSelectionMode.Box)
+                        {
+                            ReduceBoxSelectionToEditableBox(isDelete: true);
+                        }
+                        else
+                        {
+                            selection.Clear();
+                            MoveCaretToClosestEditableBuffer();
+                        }
+                    }
+                    else
+                    {
+                        var caretPosition = TextView.Caret.Position.BufferPosition;
+                        var isInActivePrompt = IsInActivePrompt(caretPosition);
+
+                        if (isInActivePrompt)
+                        {
+                            MoveCaretToClosestEditableBuffer();
+                        }
+                        else if (MapToEditableBuffer(caretPosition) == null)
+                        {
+                            return false;
+                        }
+
+                        // Move caret to the begining of the line for pasting full-line when selection is empty
+                        // If the caret was in active prompt, it was already moved to strat of input line
+                        if (dataHasLineCutCopyTag && !isInActivePrompt)
+                        {
+                            var endPoint = GetSourceBufferPoint(caretPosition.GetContainingLine().End);
+                            TextView.Caret.MoveTo(GetProjectionBufferPoint(endPoint.GetContainingLine().Start));
+                        }
+                    }
+
+                    if (dataHasBoxCutCopyTag)
+                    {
+                        if (selection.IsEmpty && IsCaretOnBlankEditableLine())
+                        {
+                            InsertText(code);
+                        }
+                        else
+                        {
+                            VirtualSnapshotPoint unusedStart, unusedEnd;
+                            EditorOperations.InsertTextAsBox(code, out unusedStart, out unusedEnd);
+                        }
+                    }
+                    else
+                    {
+                        InsertText(code);
+                    }
+
+                    // don't record undo histroy for standard-input
+                    if (!ReadingStandardInput)
+                    {
+                        transaction.Complete();
+                    }
+                    return true;
+                }
+            }
+            
+            /// <summary>Implements <see cref="IInteractiveWindowOperations.Cut"/>.</summary>
+            public void Cut()
```

Comment:
I'm quite pleased that we were able to express cut as copy+paste.  Would you mind adding a comment to that effect?